packages/rollup-contracts/test/rollup-list/StateCommitmentChain.spec.ts

```diff
@@ -0,0 +1,377 @@
+import '../setup'
+
+/* External Imports */
+import { getLogger, TestUtils } from '@eth-optimism/core-utils'
+import { createMockProvider, deployContract, getWallets } from 'ethereum-waffle'
+import { Contract } from 'ethers'
+
+/* Internal Imports */
+import { StateChainBatch } from './RLhelper'
+
+/* Logging */
+const log = getLogger('state-commitment-chain', true)
+
+/* Contract Imports */
+import * as StateCommitmentChain from '../../build/StateCommitmentChain.json'
+import * as CanonicalTransactionChain from '../../build/CanonicalTransactionChain.json'
+import * as RollupMerkleUtils from '../../build/RollupMerkleUtils.json'
+
+/* Begin tests */
+describe('StateCommitmentChain', () => {
+  const provider = createMockProvider()
+  const [
+    wallet,
+    sequencer,
+    l1ToL2TransactionPasser,
+    fraudVerifier,
+    randomWallet,
+  ] = getWallets(provider)
+  let stateChain
+  let canonicalTxChain
+  let rollupMerkleUtils
+  const DEFAULT_STATE_BATCH = ['0x1234', '0x5678']
+  const DEFAULT_TX_BATCH = [
+    '0x1234',
+    '0x5678',
+    '0x1234',
+    '0x5678',
+    '0x1234',
+    '0x5678',
+    '0x1234',
+    '0x5678',
+    '0x1234',
+    '0x5678',
+  ]
+  const DEFAULT_STATE_ROOT = '0x1234'
+  const FORCE_INCLUSION_PERIOD = 600
+
+  const appendAndGenerateBatch = async (
+    batch: string[],
+    batchIndex: number = 0,
+    cumulativePrevElements: number = 0
+  ): Promise<StateChainBatch> => {
+    await stateChain.appendStateBatch(batch)
+    // Generate a local version of the rollup batch
+    const localBatch = new StateChainBatch(
+      batchIndex,
+      cumulativePrevElements,
+      batch
+    )
+    await localBatch.generateTree()
+    return localBatch
```

Comment:
I think we have relatively good unit tests around this from the class which `StateChainBatch` extends, but a potentially good sanity check here would be to confirm that the `elementsMerkleRoot` which gets stored on-chain matches the `localBatch`'s root returned here.

Reply:
think this is solved by the "calculates Batch header hash correctly" test - hard to check if the Merkle root is calculated the same since it's not actually stored, it's immediately hashed w/ the batch header and stored as the batch header hash